pyo3-derive-backend/src/from_pyobject.rs

```diff
@@ -249,7 +249,7 @@ impl<'a> Container<'a> {
         let self_ty = &self.path;
         let mut fields: Punctuated<TokenStream, syn::Token![,]> = Punctuated::new();
         for i in 0..len {
-            fields.push(quote!(slice[#i].extract()?));
+            fields.push(quote!(s.get_item(#i).extract()?));
```

Comment:
Is the memory layout different for `PyTuple`s with limited ABI? I know this change here is necessary because `as_slice()` is no longer available on `PyTuple`:
https://github.com/PyO3/pyo3/blob/2ec1c3b0b9bb0c2e5ecd07d9014ac1a9ae890634/src/types/tuple.rs#L84-L93

It probably won't have a measurable impact anyways, just curious..

Reply:
They _can_ be, which is the whole point of the limited ABI :-)